servers/audio_server.cpp

```diff
@@ -922,6 +922,22 @@ float AudioServer::get_global_rate_scale() const {
 	return global_rate_scale;
 }
 
+void AudioServer::notify_source_playing() {
+	MutexLock lock(playing_sources_mutex);
+	if (playing_sources_count == 0) {
```

Comment:
To manage `playing_sources_count` safely you can just use it as an atomic and avoid the mutex. To do it Godot-style (by the way, #45617 will introduce a new style):
```
uint32_t playing_sources_count;

// [...]

void AudioServer::notify_source_playing() {
    if (atomic_increment(&playing_sources_count) == 1) {
        AudioDriver::get_singleton()->set_sleep_state(false);
    }
}

void AudioServer::notify_source_stopped_playing() {
    if (atomic_decrement(&playing_sources_count) == 0) {
        last_playback_time_msec = OS::get_singleton()->get_ticks_msec();
    }
}
```

Reply:
Done!